### PR TITLE
py3: remove string compat related code

### PR DESCRIPTION
**Pull request by mroutis**
Related: https://github.com/iterative/dvc/issues/1818

### CODE REVIEW

File: dvc/cache.py [hunk 1]
```diff
@@ -54,7 +53,7 @@ def getter(self):
 
         return Remote(self.repo, name=remote)
 
-    getter.__name__ = builtin_str(name)
+    getter.__name__ = str(name)
```
Comment (efiop) [inline dvc/cache.py]: Do we still need this conversion though?

File: dvc/path_info.py [hunk 2]
```diff
@@ -54,7 +51,7 @@ def __str__(self):
         return relpath(path)
 
     def __repr__(self):
-        return builtin_str("{}: '{}'").format(type(self).__name__, self)
+        return str("{}: '{}'").format(type(self).__name__, self)
```
Comment (efiop) [inline dvc/path_info.py]: This `str("")` has no sense 🙂

**Comment by mroutis** [comment]
thanks for the review, @efiop ! I'll address those comments in the next PR.

**Comment by efiop** [comment]
@mroutis Oops, sorry, merged accidentally. 🙁

**Pull request merged**